src/pyodide-py/pyodide/console.py

```diff
@@ -150,4 +218,49 @@ def banner(self):
         """ A banner similar to the one printed by the real Python interpreter. """
         # copyied from https://github.com/python/cpython/blob/799f8489d418b7f9207d333eac38214931bd7dcc/Lib/code.py#L214
         cprt = 'Type "help", "copyright", "credits" or "license" for more information.'
-        return f"Python {platform.python_version()} {platform.python_build()} on WebAssembly VM\n{cprt}"
+        version = platform.python_version()
+        build = f"({', '.join(platform.python_build())})"
+        return f"Python {version} {build} on WebAssembly VM\n{cprt}"
+
+
+def repr_abreviate(
+    value: Any, limit: int = 1000, split: Optional[int] = None, separator: str = "..."
+):
+    """Compute the string representation of `value` and abreviat it
+    if necessary.
+
+    If it is longer than `limit` then return the firsts `split`
+    characters and the last `split` characters seperated by '...'.
+    Default value for `split` is `limit // 2`.
+    """
+    if split is None:
+        split = limit // 2
+    text = repr(value)
+    if len(text) > limit:
+        text = f"{text[:split]}{separator}{text[-split:]}"
+    return text
+
+
+def displayhook(value, repr: Callable[[Any], str]):
+    """A displayhook with custom `repr` function.
+
+    It is intendend to overload `sys.displayhook`. Note that monkeypatch
+    `builtins.repr` does not work in `sys.displayhook`. The pointer to
+    `repr` seems hardcoded in default `sys.displayhook` version
+    (which is written in C)."""
+    # from https://docs.python.org/3/library/sys.html#sys.displayhook
+    if value is None:
+        return
+    builtins._ = None  # type: ignore
```

Comment:
This needs an inline comment on why we need it.

Reply:
You mean a comment on why we need displayhook? In the docstring?

Reply:
No I mean why we need to manually change `builtins._` here and the end of the function. Adding an inline comment in the code would help the reader...

Reply:
OK. I'll so this.